pypeit/scripts/ql_keck_deimos.py

```diff
@@ -0,0 +1,239 @@
+    # Run setup
+    ps_sci, _, indx = script_Utils.run_setup(science_file,
+                                              extension='', 
+                                              no_write_sorted=True)
+    # Check against existing PypeIt files
+    pypeit_files = glob.glob(os.path.join(pargs.redux_path, 'keck_deimos_*', 'keck_deimos_calib_*.pypeit'))
+    mtch = []
+    for pypeit_file in pypeit_files:
+        # Read
+        tmp_scriptU = Utilities('keck_deimos', pypeit_file=pypeit_file)
+        # Check for a match
+        match = True
+        for key in tmp_scriptU.spectrograph.configuration_keys():
+            if ps_sci.fitstbl[key][0] != tmp_scriptU.sdict[key]:
+                match = False
+        if match:
+            mtch.append(pypeit_file)
+    if len(mtch) != 1:
+        msgs.error("Matched to more than one setup.  Inconceivable!")
+
+    return mtch[0], ps_sci
+
+def run_on_science(pargs, script_Utils, calib_pypeit_file, ps_sci):
+    """
+    Process a science frame
+
+    Args:
+        pargs (argparse.ArgumentParser):
+            Command line arguments
+        script_Utils:
+        calib_pypeit_file (str):
+        ps_sci (:class:`pypeit.pypeitsetup.PypeItSetup`):
+    """
+    # Load up and slurp PypeIt file
+    ps = PypeItSetup.from_pypeit_file(calib_pypeit_file)
+    # Parse science file info
+    science_file = os.path.join(pargs.full_rawpath, pargs.science)
+    science_pypeit = calib_pypeit_file.replace('calib', 'science')
+    # Add to file list, it not present
+    if science_file not in ps.file_list:
+        ps.file_list += [science_file]
+        # Add to usrdata 
+        new_row = {}
+        for key in ps.usrdata.keys():
+            new_row[key] = ps_sci.fitstbl[key][0]
+        ps.usrdata.add_row(new_row)
+
+    # Build
+    _ = ps.build_fitstbl()
+
+    # Remove any extraneous science files in the folder
+    gd_files = (ps.fitstbl['filename'] == os.path.basename(science_file)) | (
+        ps.fitstbl['frametype'] != 'science')
+    ps.fitstbl.table = ps.fitstbl.table[gd_files]
+    ps.file_list = np.array(ps.file_list)[gd_files].tolist()
+
+    # Generate PypeIt file
+    # TODO -- Generalize this with metadata.write_pypeit()
+    # Write the file
+    setup_lines = yaml.dump(utils.yamlify(ps.setup_dict)).split('\n')[:-1]
+    with io.StringIO() as ff:
+        ps.usrdata.write(ff, format='ascii.fixed_width')
+        data_lines = ff.getvalue().split('\n')[:-1]
+    paths = [os.path.dirname(item) for item in ps.file_list]
+    paths = np.unique(paths)
+
+    # Add to configs
+    if pargs.slit_spat is not None:
+        # Remove detnum
+        for kk, item in enumerate(ps.user_cfg):
+            if 'detnum' in item:
+                ps.user_cfg.pop(kk)
+        ridx = ps.user_cfg.index('[rdx]')
+        ps.user_cfg.insert(ridx+1, '    slitspatnum = {0}'.format(pargs.slit_spat))
+    else:
+        embed(header='NOT READY:  118 of ql_deimos')
```

Comment:
Any reason in particular to use an embed here? If we stick with it, updating the line number would be nice. We could also head this error off by exiting if `calib_only is None and slit_spat is not None` in `main()`, since there isn't really a use case for running QL on every slit.

Reply:
good point, although I've changed to a `NotImplementedError`